observatory-platform/observatory/platform/workflows/stream_telescope.py

```diff
@@ -268,19 +268,19 @@ def bq_delete_old(self, release: StreamRelease, **kwargs):
 
         dag_run: DagRun = kwargs["dag_run"]
         ti: TaskInstance = kwargs["ti"]
-        start_date = pendulum.instance(get_prev_start_date_success_task(dag_run, ti.task_id))
-        end_date = release.end_date
-        if (end_date - start_date).days + 1 >= self.bq_merge_days:
+        start_date = pendulum.instance(get_prev_start_date_success_task(dag_run, ti.task_id)).start_of("day")
```

Comment:
Should `start_date =` be using the start date of the previous task or the `release.end_date` of the previous release that was merged? Since the partition date used by `bq_load_ingestion_partition` is `release.end_date`?

Also, in the SQL queries `insert_from_partitions` and `merge_delete_matched`, `_PARTITIONDATE > '{{ start_date }}' AND _PARTITIONDATE <= '{{ end_date }}'` will always skip the start date? 

And the start_date supplied to the query will always skip partitions because the partition is set to `release.end_date`, which always occurs before the date that the previous task ran.

I wonder if it would make it more robust to query the BigQuery table to find out the last partition date? It is probably the best source of truth for that information in this case because the BigQuery table is what is being updated.